src/server/pfs/server/trigger.go

```diff
@@ -21,72 +20,48 @@ func (d *driver) triggerCommit(
 	commit *pfs.Commit,
 ) error {
 	repoInfo := &pfs.RepoInfo{}
-	if err := d.repos.ReadWrite(txnCtx.SqlTx).Get(commit.Branch.Repo, repoInfo); err != nil {
+	if err := d.repos.ReadWrite(txnCtx.SqlTx).Get(commit.Repo, repoInfo); err != nil {
 		return errors.EnsureStack(err)
 	}
-	commitInfo := &pfs.CommitInfo{}
-	if err := d.commits.ReadWrite(txnCtx.SqlTx).Get(commit, commitInfo); err != nil {
+	newHead := &pfs.CommitInfo{}
+	if err := d.commits.ReadWrite(txnCtx.SqlTx).Get(commit, newHead); err != nil {
 		return errors.EnsureStack(err)
 	}
-
-	// Track any branches that are triggered and their new (alias) head commit
-	triggeredBranches := map[string]*pfs.CommitInfo{}
-	triggeredBranches[commitInfo.Commit.Branch.Name] = commitInfo
-
-	var triggerBranch func(branch *pfs.Branch) (*pfs.CommitInfo, error)
-	triggerBranch = func(branch *pfs.Branch) (*pfs.CommitInfo, error) {
-		if newHead, ok := triggeredBranches[branch.Name]; ok {
-			return newHead, nil
-		}
-
+	for _, b := range repoInfo.Branches {
```

Comment:
I haven't thought about this before, but this could have some concerning scaling characteristics. It is m * n, where m is the number of commits and n is the number of branches. I don't think we should try and address that in this PR, I just thought it was worth noting.